### DISCUSSION TITLE
Planning List Variale has achieved real-time planning?

### DISCUSSION THREAD

**Discussion by kentbill** [TimefoldAI/timefold-solver#904] [General]
In the document about Planning list variables, I found that if I need to use real-time planning, I no longer need to use the Chained Through Time pattern. Does it mean that Planning list variables support real-time planning now? If so, can I use the ProblemChanged interface to submit a modified Planning list variable directly?
[7. Planning list variable (VRP, Task assigning, …​)](https://docs.timefold.ai/timefold-solver/latest/using-timefold-solver/modeling-planning-problems#planningListVariable)

**Comment by ge0ffrey** [comment]
Both list vars and chained vars support Real-Time Planning.

That being said, list variables are typically much cleaner than chained variables.

**Reply by kentbill** [reply]
Thank you! Geoffrey.
